src/frontend/components/UI/AuthModal/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useRef } from 'react'
+import styles from './index.module.scss'
+import { ModalAnimation } from '@hyperplay/ui'
+import { WebviewTag } from 'electron'
+import { observer } from 'mobx-react-lite'
+import authModalState from '../../../state/authModalState'
+import { ethers } from 'ethers'
+import extensionStore from '../../../store/ExtensionStore'
+import onboardingStore from '../../../store/OnboardingStore'
+
+// TODO: replace this with dev portal prod URL when it's ready
+const url =
+  'https://hyperplay-dev-git-feature-unified-auth-ui-valist.vercel.app/signin'
+
+const METAMASK_ALREADY_PROVIDED_ERROR_CODE = -32002
+
+const isTooManyRequestsError = (error: string) => {
+  return error.includes(METAMASK_ALREADY_PROVIDED_ERROR_CODE.toString())
+}
+
+const AuthModal = () => {
+  const webviewRef = useRef<WebviewTag>(null)
+
+  const handleAccountNotConnected = async () => {
```

Comment:
after calling the flow I detailed below, you could return a promise and when `window.api.handleAccountsChanged(cb)` calls cb, you can be sure it's connected and resolve. then you can go on and resume the auth modal connection stuff

Reply:
actually you could probably use `WalletStore.address` and let mobx trigger a re-render. that's a lot cleaner